components/dashboard/src/components/WorkspaceClassesOptions.tsx

```diff
@@ -0,0 +1,224 @@
+                    onClose();
+                },
+            },
+        );
+    };
+
+    const computedError = useMemo(() => {
+        const leftOptions =
+            allowedClasses.filter((c) => !c.isDisabledInScope && !restrictedClasses.includes(c.id)) ?? [];
+        if (leftOptions.length === 0) {
+            return "At least one workspace class has to be selected.";
+        }
+        if (!defaultClass || !leftOptions.find((cls) => cls.id === defaultClass)) {
+            return "A default workspace class is required.";
+        }
+        return;
+    }, [restrictedClasses, allowedClasses, defaultClass]);
+
+    return (
+        <Modal visible onClose={onClose} onSubmit={handleUpdate}>
+            <ModalHeader>Available workspace classes</ModalHeader>
+            <ModalBody>
+                {allowedClasses.map((wsClass) => (
+                    <WorkspaceClassSwitch
+                        showSetDefaultButton={showSetDefaultButton}
+                        restrictedClasses={restrictedClasses}
+                        wsClass={wsClass}
+                        isDefault={defaultClass === wsClass.id}
+                        checked={!restrictedClasses.includes(wsClass.id)}
+                        onSetDefault={() => {
+                            setDefaultClass(wsClass.id);
+                        }}
+                        onCheckedChange={(checked) => {
+                            const newVal = !checked
+                                ? restrictedClasses.includes(wsClass.id)
+                                    ? [...restrictedClasses]
+                                    : [...restrictedClasses, wsClass.id]
+                                : restrictedClasses.filter((id) => id !== wsClass.id);
+                            setRestrictedClasses(newVal);
+                        }}
+                    />
+                ))}
+            </ModalBody>
+            <ModalBaseFooter className="justify-between">
+                <div className="text-red-500">
+                    {(computedError || updateMutation.isError) && (computedError || String(updateMutation.error))}
+                </div>
+                <div className="flex gap-2">
+                    <Button variant="secondary" onClick={onClose}>
+                        Cancel
+                    </Button>
+                    <LoadingButton disabled={!!computedError} type="submit" loading={updateMutation.isLoading}>
+                        Save
+                    </LoadingButton>
+                </div>
+            </ModalBaseFooter>
+        </Modal>
+    );
+};
+
+const WorkspaceClassSwitch = ({
+    wsClass,
+    checked,
+    isDefault,
+    onCheckedChange,
+    ...props
+}: {
+    wsClass: AllowedWorkspaceClass;
+    restrictedClasses: string[];
+    checked: boolean;
+    isDefault: boolean;
+    showSetDefaultButton?: boolean;
+    onSetDefault: () => void;
+    onCheckedChange: (checked: boolean) => void;
```

Comment:
Could we keep this consistent and define a `Prop` type here?

Reply:
@filiptronicek  Addressed. How we define 'consistent'? For every component need a Props or once keys of props more than x items?

Reply:
Good point. I try to do it everywhere, even just for one prop, but I can understand that one would find that excessive. I think if it makes the function signature easier to read and mentally parse, it's worth doing.

Reply:
OK, I'll make sure my followed dashboard changes align to it 👍